src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/bubbleVisualization.js

```diff
@@ -0,0 +1,40 @@
+class BubbleVisualization extends Visualization {
+    static inputType = "Any"
+
+    onDataReceived(data) {
+        this.setPreprocessor("None");
+
+        const xmlns = "http://www.w3.org/2000/svg";
+        while (this.dom.firstChild) {
+            this.dom.removeChild(this.dom.lastChild);
+        }
+        const width = this.dom.getAttributeNS(null, "width");
+        const height = this.dom.getAttributeNS(null, "height");
+
+        const svgElem = document.createElementNS(xmlns, "svg");
+        svgElem.setAttributeNS(null, "id"     , "vis-svg");
+        svgElem.setAttributeNS(null, "viewBox", 0 + " " + 0 + " " + width + " " + height);
+        svgElem.setAttributeNS(null, "width"  , "100%");
+        svgElem.setAttributeNS(null, "height" , "100%");
+        svgElem.setAttributeNS(null, "transform", "matrix(1 0 0 -1 0 0)");
+
+        this.dom.appendChild(svgElem);
+
+        data.forEach(data => {
+            const bubble = document.createElementNS(xmlns,"circle");
+            bubble.setAttributeNS(null,"stroke", "black");
+            bubble.setAttributeNS(null,"fill"  , "red");
+            bubble.setAttributeNS(null,"r"     , data[2]);
+            bubble.setAttributeNS(null,"cx"    , data[0]);
+            bubble.setAttributeNS(null,"cy"    , data[1]);
```

Comment:
I know its not yours code, but lets impriove it. Default to something instead of crashing